klemens_basic_thoughts_1.md

```diff
@@ -0,0 +1,206 @@
+# Design Proposal
+
+# Concepts
+
```

Comment:
So my guess is that these concepts are correct and reasonable, but I"m very afraid to go down this path.  First, we have no compiler/stdlib to really test them against.  Also boost.process isn't specified this way.  And last, the Library groups are oddly concerned about specifications with Concepts -- for the standard library the issue is that any change to a concept once published will result in code breakage downstream.  Anyway, I'm thinking maybe we should put this as one of the questions to LEWGI -- do they want to see this library specified with concepts?

Reply:
We can use the inheritance version currently in use. 
I think concepts would be the better fit overall, but htat is something we can change later on.

Reply:
After processing comment below for awhile it seems like we should use concepts.  Really after c++20 I would think that should be the baseline for new library proposals.